Support file chunking and add support for s2 coverings

ThomasThelen opened the pull request:
Couple of changes here

## Directory Changes

I broke quite a bit of the code up into core library files. These are shared between the s2 and integration command line tools.

## Support for S2 over a geometry

I never implemented the feature to generate s2 cells over a geometry. This PR has that change in the form of the `S2OverlapGenerator` class

## Chunking Support

To fix #13 I added support for chunking rdf output. The default is 100,000. Meaning, what was once 100,000 separate files written to disk are written as a single file. I was able to test this with level 13, which wrote around 300 files for a total of almost 400GB of data

## Removed config

The config is sort of difficult to work with because it can't be modified through the docker image. Instead, I moved the config parameters to cli arguments

ckfisher reviewed (commented):
I'm good with everything here for the most part and was able to run things fine. A few nitpicks items though: 

1. The "Integrating S2 Cells With.." section of the readme has a code snippet of just kwg-int:sfwithin twice. I don't get why this is here.
2. I noticed in the argument help definitions of integrate.py a lot were listed as Unknown even though they are defined and defaults set in integrator.py (line 32). I had to dig to find the tolerance, but it is `tolerance (float, optional): maximal segment width. Defaults to 1e-2.` in geometric_feature.py. 
3. What happens in geometryParser.parse if it is passed invalid geometries? Should there be error check in some of these methods? I guess this is probably true for most of the code, as I don't see a lot of error checking or messaging. Future feature request?

ThomasThelen commented:
1. Fixed in [8dbc4ec](https://github.com/KnowWhereGraph/s2-coverings/pull/15/commits/8dbc4ecf8f861bf253e47b5f33c3f939e417a3c3). Good catch.
2. Added descriptions in [e1c5abd](https://github.com/KnowWhereGraph/s2-coverings/pull/15/commits/e1c5abdf6a907860292bd471186f8506b1fe83f2) and filed https://github.com/KnowWhereGraph/s2-coverings/issues/18 during testing
3. Filed an issue for enhancing error checking. https://github.com/KnowWhereGraph/s2-coverings/issues/17
